src/components/editableTable/SingleEditableTableRow.js

```diff
@@ -133,69 +106,43 @@ export default function EditableTableRow({
         }
     };
 
-    const Cell = column => {
-        const Content = () =>
-            (editing && column.editable && editable) || (isNewRow && editing && column.required)
-                ? inputComponentFactory(item, column, handleValueChange, rest)
-                : displayComponentFactory(item, column);
-        if (!column.tooltip) {
-            return (
-                <Fragment key={`${column?.id}${item.id}`}>
-                    <TableCell id={column.type}>
-                        <>{Content()}</>
-                    </TableCell>
-                </Fragment>
-            );
-        }
-        return (
-            <Fragment key={`${column?.id}${item.id}`}>
-                <Tooltip title={column.tooltip(item) || ''}>
-                    <TableCell id={column.type} key={`${column?.id}${item.id}`}>
-                        {Content()}
-                    </TableCell>
-                </Tooltip>
-            </Fragment>
-        );
-    };
-
-    Cell.propTypes = {
-        column: PropTypes.shape({
-            tooltip: PropTypes.func,
-            id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-            editable: PropTypes.bool,
-            type: PropTypes.string,
-            required: PropTypes.bool
-        })
-    };
-
-    Cell.defaultProps = {
-        column: null
+    const handleEditButtonClick = () => {
+        setEditing(true);
     };
 
     return (
         <ClickAwayListener onClickAway={e => handleClickAway(e)}>
-            <TableRow>
-                {columns.map(column => Cell(column))}
+            <TableRow onClick={!editing && !deleteRowPreEdit && handleEditButtonClick}>
+                {columns.map(column => (
+                    <EditableTableCell
```

Comment:
won't these need key={column.id} or similar to appease the react unique key prop gods? Or is it expecting to find a key prop in {...rest} ?